semgrep/semgrep/output.py

```diff
@@ -226,8 +228,24 @@ def __init__(
         self.final_error: Optional[Exception] = None
 
     def handle_semgrep_errors(self, errors: List[SemgrepError]) -> None:
+        timeout_errors = defaultdict(list)
         for err in errors:
-            self.handle_semgrep_error(err)
+            if isinstance(err, MatchTimeoutError):
```

Comment:
Do we run into issues since codepaths diverge now? i.e. when the error is a MatchTimeoutError we don't set self.has_output to True, we don't check membership in self.error_set before appending to semgrep_structured_errors etc. (see handle_semgrep_error)

Reply:
Oh I missed that, let me push an edit to this file...